src/python/pants/backend/python/lint/bandit/rules_integration_test.py

```diff
@@ -3,57 +3,50 @@
 
 from typing import List, Optional
 
-from pants.backend.python.lint.bandit.rules import BanditLinter
+from pants.backend.python.lint.bandit.rules import BanditConfiguration, BanditConfigurations
 from pants.backend.python.lint.bandit.rules import rules as bandit_rules
-from pants.backend.python.targets.python_library import PythonLibrary
+from pants.backend.python.rules.targets import PythonInterpreterCompatibility, PythonLibrary
 from pants.base.specs import FilesystemLiteralSpec, OriginSpec, SingleAddress
 from pants.build_graph.address import Address
-from pants.build_graph.build_file_aliases import BuildFileAliases
-from pants.engine.fs import FileContent, InputFilesContent, Snapshot
-from pants.engine.legacy.structs import PythonTargetAdaptor, PythonTargetAdaptorWithOrigin
+from pants.engine.fs import FileContent
 from pants.engine.rules import RootRule
 from pants.engine.selectors import Params
+from pants.engine.target import TargetWithOrigin
 from pants.rules.core.lint import LintResult
-from pants.source.wrapped_globs import EagerFilesetWithSpec
 from pants.testutil.option.util import create_options_bootstrapper
 from pants.testutil.test_base import TestBase
 
 
 class BanditIntegrationTest(TestBase):
 
-    good_source = FileContent(path="test/good.py", content=b"hashlib.sha256()\n")
+    good_source = FileContent(path="good.py", content=b"hashlib.sha256()\n")
     # MD5 is a insecure hashing function
-    bad_source = FileContent(path="test/bad.py", content=b"hashlib.md5()\n")
-
-    @classmethod
-    def alias_groups(cls) -> BuildFileAliases:
-        return BuildFileAliases(targets={"python_library": PythonLibrary})
+    bad_source = FileContent(path="bad.py", content=b"hashlib.md5()\n")
 
     @classmethod
     def rules(cls):
-        return (*super().rules(), *bandit_rules(), RootRule(BanditLinter))
+        return (*super().rules(), *bandit_rules(), RootRule(BanditConfigurations))
 
     def make_target_with_origin(
         self,
         source_files: List[FileContent],
         *,
         interpreter_constraints: Optional[str] = None,
         origin: Optional[OriginSpec] = None,
-    ) -> PythonTargetAdaptorWithOrigin:
-        input_snapshot = self.request_single_product(Snapshot, InputFilesContent(source_files))
-        adaptor_kwargs = dict(
-            sources=EagerFilesetWithSpec("test", {"globs": []}, snapshot=input_snapshot),
-            address=Address.parse("test:target"),
+    ) -> TargetWithOrigin:
+        for source_file in source_files:
+            self.create_file(source_file.path, source_file.content.decode())
+        target = PythonLibrary(
+            {PythonInterpreterCompatibility.alias: interpreter_constraints},
+            address=Address.parse(":target"),
         )
```

Comment:
NB: this will use the default value for everything else, including the default value for `sources`.